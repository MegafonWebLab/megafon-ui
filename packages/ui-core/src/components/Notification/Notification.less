@import '~styles/base';

@b: e('.mfui-beta-notification');

@{b} {
    position: relative;

    padding: 16px 12px;

    @media @mobileBU {
        padding: 20px 28px;
    }
    @media @desktop {
        padding: 20px 32px;
    }

    &__container {
        display: flex;
    }

    &__title {
        margin-bottom: 8px;

        @media @desktopMU {
            margin-bottom: 4px;
        }
    }

    &__title_close-padding {
        padding-right: 20px;
    }

    &__text {
        .commonFont();
        max-width: 100%;
        margin: 0;
    }

    &__text_close-padding {
        padding-right: 20px;
    }

    &__content {
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        align-self: center;
        width: 100%;
        max-width: 212px;
        margin-left: 12px;

        @media @mobileBU {
            max-width: 540px;
        }
        @media @desktop {
            max-width: 640px;
        }
        @media @desktopMU {
            max-width: 840px;
        }
    }

    &__icon-container {
        display: flex;
        flex-shrink: 0;
        align-items: center;
        align-self: flex-start;
        justify-content: center;
        width: 40px;
        min-width: 40px;
        height: 40px;
        border-radius: 50%;
    }

    &__attention-icon,
    &__icon-container svg {
        width: 32px;
        min-width: 32px;
        height: 32px;
    }

    &__link {
        .h5();
        display: inline-flex;
        align-items: center;
        margin-top: 8px;

        cursor: pointer;
    }

    &__right-arrow {
        width: 20px;
        min-width: 20px;
        height: 20px;
        margin-top: 4px;

        fill: @link;
    }

    &__close {
        position: absolute;
        top: 12px;
        right: 12px;

        display: flex;
        padding: 0;
        border: none;
        border-radius: 50%;

        background: transparent;
        cursor: pointer;
    }

    &__close:hover {
        background-color: @spbSky160;
    }

    &__close-icon {
        width: 20px;
        height: 20px;

        fill: @freshAsphalt;

        @media @mobileBU {
            width: 32px;
            height: 32px;
        }
    }

    &_type_error {
        @{b} {
            &__icon-container {
                background-color: @warmRedC20;
            }

            &__icon-container svg {
                fill: @warmRedC;
            }
        }
    }

    &_type_success {
        @{b} {
            &__icon-container {
                background-color: @green20;
            }

            &__icon-container svg {
                fill: @green;
            }
        }
    }

    &_type_warning {
        @{b} {
            &__icon-container {
                background-color: @mandarin20;
            }

            &__icon-container svg {
                fill: @mandarin;
            }
        }
    }

    &_type_info {
        @{b} {
            &__icon-container {
                background-color: @Pantone311C20;
            }

            &__icon-container svg {
                fill: @Pantone311C;
            }
        }
    }

    &_colored {
        @{b} {
            &__icon-container {
                background-color: @clearWhite;
            }

            &__close:hover {
                background-color: @clearWhite60;
            }
        }
    }

    &_colored&_type_error {
        background-color: @error;

        @{b} {
            &__title,
            &__text,
            &__link {
                color: @clearWhite;
            }

            &__right-arrow {
                fill: @clearWhite;
            }

            &__icon-container {
                background-color: @clearWhite20;
            }

            &__icon-container svg {
                fill: @clearWhite;
            }
        }
    }

    &_colored&_type_success {
        background-color: @success;

        @{b}__icon-container svg {
            fill: @green;
        }
    }

    &_colored&_type_warning {
        background-color: @warning;

        @{b}__icon-container svg {
            fill: @mandarin;
        }
    }

    &_colored&_type_info {
        background-color: @info;

        @{b}__icon-container svg {
            fill: @Pantone311C;
        }
    }
}
