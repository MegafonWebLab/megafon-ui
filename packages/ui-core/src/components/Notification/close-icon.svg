<svg viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg">
    <path fill-rule="evenodd" clip-rule="evenodd" d="M5 3.75 1.25 0 0 1.25 3.75 5 0 8.75 1.25 10 5 6.25 8.75 10 10 8.75 6.25 5 10 1.25 8.75 0 5 3.75Z"/>
</svg>
