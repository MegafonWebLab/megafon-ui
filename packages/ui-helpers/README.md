# @megafon/ui-helpers

Helpers library

## Installation

with npm
```bash
npm install --save @megafon/ui-helpers@beta
```

or yarn
```bash
yarn add @megafon/ui-helpers@beta
```

## Usage

---
### cnCreate
BEM generator

```js
import { cnCreate } from '@megafon/ui-helpers';

const cn = cnCreate('block');

cn();
// block

cn('element');
// block__element

cn({ mod: true });
// block block_mod

cn({ mod: 'value' });
// block block_mod_value

cn(['custom-class', 'custom-class-2']);
// block custom-class custom-class-2

cn('element', ['custom-class', 'custom-class-2']);
// block__element custom-class custom-class-2

cn('element', { mod: true }, ['custom-class', 'custom-class-2']);
// block__element block__element_mod custom-class custom-class-2
```

---
### convert
htmr library extension for replacing html tags with your own components, recognizing
components props and html attributes

requires React as dependency

```tsx
import React from 'react';
import { convert, ConvertTransformConfig } from '@megafon/ui-helpers';

const Link = ({ href, target, className, children }) => (
    <a className={className} href={href} target={target}>{children}</a>
);

const config: ConvertTransformConfig = {
    a: {
        component: Link,
        props: ['href', 'target'],
        customProps: { className: 'class-name' },
    },
};

convert('<a href="/test" target="_blank">link</a>', config);
// <Link className="class-name" href="/test" target="_blank">link</Link>'
```
---

### filterDataAttrs
Filters properties using regular expression /^data-/

```ts
import { filterDataAttrs } from '@megafon/ui-helpers';

filterDataAttrs({
    'data-attr': 'value',
    'data-custom-attr': 'value',
    'DataAttr': 'value',
    'attr-data': 'value'
});
// { 'data-attr': 'value', 'data-custom-attr': 'value' }
```
---

### detectTouch
Detects touch devices

```js
import { detectTouch } from '@megafon/ui-helpers';

detectTouch()
// returns `true` or `false` depending on device

```
---
