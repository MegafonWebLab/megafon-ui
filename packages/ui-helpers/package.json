{
  "name": "@megafon/ui-helpers",
  "version": "2.4.0",
  "main": "dist/lib/index.js",
  "module": "dist/es/index.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/MegafonWebLab/megafon-ui.git"
  },
  "author": "MegaFon",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "cleanDist": "rm -rf dist",
    "coverage": "yarn coverage:unit",
    "coverage:unit": "jest --ci --collectCoverage --coverageDirectory=coverage",
    "typecheck": "tsc --noEmit -p .",
    "prettier:check": "prettier --list-different \"src/**/*.{js,jsx,ts,tsx}\"",
    "prettier:fix": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "lint": "yarn lint:ts",
    "lint:ts": "eslint --ext .js,.jsx,.ts,.tsx src --max-warnings=0",
    "test": "yarn test:unit",
    "test:unit": "jest",
    "test:update": "jest --updateSnapshot"
  },
  "publishConfig": {
    "access": "public"
  },
  "lint-staged": {
    "src/**/**/*.(js|jsx|ts|tsx)": [
      "eslint --max-warnings=0 --config ../../.eslintrc.json"
    ]
  },
  "peerDependencies": {
    "react": ">=16.13.0"
  },
  "dependencies": {
    "@babel/runtime": "^7.8.4",
    "classnames": "^2.2.6",
    "core-js": "^3.6.4",
    "htmr": "^0.9.2"
  },
  "devDependencies": {
    "@babel/core": "^7.8.6",
    "@babel/plugin-transform-runtime": "^7.8.3",
    "@babel/preset-env": "^7.8.6",
    "@babel/preset-react": "^7.8.3",
    "@babel/preset-typescript": "^7.8.3",
    "@rollup/plugin-babel": "^5.3.0",
    "@rollup/plugin-commonjs": "^19.0.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "@types/enzyme": "^3.10.5",
    "@types/enzyme-adapter-react-16": "^1.0.6",
    "@types/jest": "^25.1.3",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.2",
    "enzyme-to-json": "^3.4.4",
    "jest": "^25.1.0",
    "react": "^16.13.0",
    "rollup": "^2.51.1",
    "rollup-plugin-delete": "^2.0.0",
    "rollup-plugin-dts": "^3.0.2",
    "typescript": "^4.3.4"
  }
}
