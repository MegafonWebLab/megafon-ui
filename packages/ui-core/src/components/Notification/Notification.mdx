---
name: Notification
menu: Компоненты
route: components/Notification
---

import { Playground, Props } from 'docz';
import Notification from './Notification';
import { marginBottom } from './Notification.docz';

# Notification

## Properties

<Props of={Notification} />

## Basic types

<Playground>
    <div style={marginBottom}>
        <Notification type="success">Тариф «Включайся! Премиум» успешно подключен. Вторая строка показана для примера, значок выравнивается по верхней строке.</Notification>
     </div>
     <div style={marginBottom}>
        <Notification type="warning">Обратите внимание, что подключённые на текущем тарифе пакеты минут, SMS и интернета «сгорят» и не будут действовать на новом тарифе.</Notification>
     </div>
    <Notification type="error">Вы превысили количество допустимых запросов в сутки. Пожалуйста, повторите попытку позже. Строка для двух строк, выравнивание  иконки по верхней.</Notification>
</Playground>

## Short types

<Playground>
    <div style={marginBottom}>
        <Notification type="success" isShort>Success</Notification>
    </div>
    <div style={marginBottom}>
        <Notification type="warning" isShort>Warning</Notification>
    </div>
    <Notification type="error" isShort>Error</Notification>
</Playground>
