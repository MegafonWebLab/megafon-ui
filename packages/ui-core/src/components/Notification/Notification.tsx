import React from 'react';
import * as PropTypes from 'prop-types';
import './Notification.less';
import Paragraph from 'components/Paragraph/Paragraph';
import cnCreate from 'utils/cnCreate';
import AttentionIcon from 'icons/System/24/Info_24.svg';
import AttentionInvertIcon from 'icons/System/24/Info_invert_24.svg';
import CheckedIcon from 'icons/System/24/Checked_24.svg';

export const NotificationTypes = {
    SUCCESS: 'success',
    WARNING: 'warning',
    ERROR: 'error',
} as const;

type NotificationTypesType = typeof NotificationTypes[keyof typeof NotificationTypes];

export interface INotificationProps {
    /** Дополнительный класс корневого элемента */
    className?: string;
    /** Тип отображения */
    type?: NotificationTypesType;
    /** Уменьшенный размер */
    isShort?: boolean;
}

const cn = cnCreate('mfui-beta-notification');
const Notification: React.FC<INotificationProps> = ({ type = 'success', isShort = false, children, className }) => {
    const renderIcon = (): JSX.Element => {
        if (type === NotificationTypes.SUCCESS) {
            return <CheckedIcon className={cn('icon')} />;
        }

        return isShort
        ? <AttentionInvertIcon className={cn('icon')} />
        : <AttentionIcon className={cn('icon')} />;
    };

    return (
        <div
            className={cn({
                type,
                short: isShort,
            }, className)}
        >
            <div className={cn('container')}>
                {renderIcon()}
                <Paragraph
                    className={cn('text')}
                    hasMargin={false}
                >
                    {children}
                </Paragraph>
            </div>
        </div>
    );
};

Notification.propTypes = {
    type: PropTypes.oneOf(Object.values(NotificationTypes)),
    isShort: PropTypes.bool,
};

export default Notification;
