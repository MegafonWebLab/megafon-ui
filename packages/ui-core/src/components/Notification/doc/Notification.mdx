---
name: Notification
menu: Компоненты
route: components/Notification
---

import { Props } from 'docz';
import Notification from '../Notification';
import Example from './Notification.example.mdx';

# Notification

<CopyToClipboardBox text="import { Notification } from '@megafon/ui-core';" />

<DoczTabs>
    <Example />
    <Props of={Notification} />
</DoczTabs>
