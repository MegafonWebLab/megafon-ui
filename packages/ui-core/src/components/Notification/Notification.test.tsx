import * as React from 'react';
import { cnCreate } from '@megafon/ui-helpers';
import { fireEvent, render } from '@testing-library/react';
import Notification, { INotificationProps, testIdPrefix } from './Notification';

const dataAttrs: INotificationProps['dataAttrs'] = {
    root: {
        'data-testid': 'root',
    },
    close: {
        'data-testid': 'close',
    },
    text: {
        'data-testid': 'text',
    },
    title: {
        'data-testid': 'title',
    },
    collapseButton: {
        'data-testid': 'collapseButton',
    },
    button: {
        'data-testid': 'button',
    },
    link: {
        'data-testid': 'link',
    },
};

const cn = cnCreate('mfui-notification');

describe('<Notification />', () => {
    it('should render with default props', () => {
        const { container } = render(<Notification>Some text</Notification>);
        expect(container).toMatchSnapshot();
    });

    it('should render with className props', () => {
        const { getByTestId } = render(
            <Notification className="testClass" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass('testClass');
    });

    it('should render with classes props', () => {
        const { getByTestId } = render(
            <Notification
                dataAttrs={dataAttrs}
                classes={{ root: 'rootTestClass', container: 'containerTestClass', content: 'contentTestClass' }}
            >
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass('rootTestClass');
        expect(getByTestId(`${testIdPrefix}-container`)).toHaveClass('containerTestClass');
        expect(getByTestId(`${testIdPrefix}-content`)).toHaveClass('contentTestClass');
    });

    it('should render with props type="info" and isColored="true"', () => {
        const { getByTestId } = render(<Notification dataAttrs={dataAttrs}>Some text</Notification>);

        expect(getByTestId('root')).toHaveClass(cn({ type: 'info', colored: true }));
    });

    it('should render with props type="info" and isColored="false"', () => {
        const { getByTestId } = render(
            <Notification isColored={false} dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'info', colored: false }));
    });

    it('should render with props type="success" and isColored="false"', () => {
        const { getByTestId } = render(
            <Notification type="success" isColored={false} dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'success', colored: false }));
    });

    it('should render with props type="success" and isColored="true"', () => {
        const { getByTestId } = render(
            <Notification type="success" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'success', colored: true }));
    });

    it('should render with props type="warning" and isColored="true"', () => {
        const { getByTestId } = render(
            <Notification type="warning" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'warning', colored: true }));
    });

    it('should render with props type="warning" and isColored="false"', () => {
        const { getByTestId } = render(
            <Notification type="warning" isColored={false} dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'warning', colored: false }));
    });

    it('should render with props type="error" and isColored="true"', () => {
        const { getByTestId } = render(
            <Notification type="error" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'error', colored: true }));
    });

    it('should render with props type="error" and isColored="false"', () => {
        const { getByTestId } = render(
            <Notification type="error" isColored={false} dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toHaveClass(cn({ type: 'error', colored: false }));
    });

    it('should render with close button', () => {
        const { getByTestId } = render(
            <Notification hasCloseButton dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('close')).toMatchSnapshot();
        expect(getByTestId('text')).toHaveClass(cn('text', { 'close-padding': true }));
    });

    it('should render with close button and title', () => {
        const { getByTestId } = render(
            <Notification title="Title" hasCloseButton dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('close')).toBeInTheDocument();
        expect(getByTestId('text')).not.toHaveClass(cn('text', { 'close-padding': true }));
    });

    it('should render with title', () => {
        const { getByTestId } = render(
            <Notification title="Title" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('title')).toMatchSnapshot();
    });

    it('should render with shortText', () => {
        const { getByTestId } = render(
            <Notification shortText="Short text" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('text')).toMatchSnapshot('text block');
        expect(getByTestId(`${testIdPrefix}-bottom`)).toMatchSnapshot('bottom block');
    });

    it('should render with closeCollapseTitle', () => {
        const { getByText } = render(
            <Notification shortText="Short text" closeCollapseTitle="Close collapse title">
                Some text
            </Notification>,
        );

        expect(getByText('Close collapse title')).toBeInTheDocument();
    });

    it('should render with openCollapseTitle', () => {
        const { getByText } = render(
            <Notification isCollapseOpened shortText="Short text" openCollapseTitle="Open collapse title">
                Some text
            </Notification>,
        );

        expect(getByText('Open collapse title')).toBeInTheDocument();
    });

    it('should render with isCollapseOpened="true"', () => {
        const { getByText } = render(
            <Notification isCollapseOpened shortText="Short text">
                Some text
            </Notification>,
        );

        expect(getByText('Свернуть')).toBeInTheDocument();
    });

    it('should render with buttonText', () => {
        const { getByTestId } = render(<Notification buttonText="Button text">Some text</Notification>);

        expect(getByTestId(`${testIdPrefix}-bottom`)).toMatchSnapshot();
    });

    it('should render with buttonLoader', () => {
        const { getByTestId } = render(
            <Notification buttonText="Button text" buttonLoader dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('button')).toHaveClass('mfui-button_loading');
    });

    it('should render with buttonDisable', () => {
        const { getByTestId } = render(
            <Notification buttonText="Button text" buttonDisable dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('button')).toHaveClass('mfui-button_disabled');
    });

    it('should render with link and link arguments', () => {
        const { getByTestId } = render(
            <Notification link="Link" rel="contact" href="/link" target="_blank" dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        expect(getByTestId('link')).toHaveAttribute('href', '/link');
        expect(getByTestId('link')).toHaveAttribute('rel', 'contact');
        expect(getByTestId('link')).toHaveAttribute('target', '_blank');
        expect(getByTestId(`${testIdPrefix}-bottom`)).toMatchSnapshot();
    });

    it('should render with dataAttrs', () => {
        const { getByTestId, rerender } = render(
            <Notification link="Link" title="Title" hasCloseButton dataAttrs={dataAttrs} buttonText="Button text">
                Some text
            </Notification>,
        );

        expect(getByTestId('root')).toBeInTheDocument();
        expect(getByTestId('title')).toBeInTheDocument();
        expect(getByTestId('text')).toBeInTheDocument();
        expect(getByTestId('close')).toBeInTheDocument();
        expect(getByTestId('link')).toBeInTheDocument();
        expect(getByTestId('button')).toBeInTheDocument();

        rerender(
            <Notification dataAttrs={dataAttrs} shortText="Short text">
                Some text
            </Notification>,
        );

        expect(getByTestId('collapseButton')).toBeInTheDocument();
    });

    it('should call onClose', () => {
        const onCloseMock = jest.fn();
        const { getByTestId } = render(
            <Notification hasCloseButton onClose={onCloseMock} dataAttrs={dataAttrs}>
                Some text
            </Notification>,
        );

        fireEvent.click(getByTestId('close'));

        expect(onCloseMock).toBeCalled();
    });

    it('should call onLinkClick', () => {
        const onLinkClickMock = jest.fn();
        const { getByTestId } = render(
            <Notification link="/link" dataAttrs={dataAttrs} onLinkClick={onLinkClickMock}>
                Some text
            </Notification>,
        );

        fireEvent.click(getByTestId('link'));

        expect(onLinkClickMock).toBeCalled();
    });

    it('should call onButtonClick', () => {
        const onButtonClickMock = jest.fn();
        const { getByTestId } = render(
            <Notification dataAttrs={dataAttrs} buttonText="Button text" onButtonClick={onButtonClickMock}>
                Some text
            </Notification>,
        );

        fireEvent.click(getByTestId('button'));

        expect(onButtonClickMock).toBeCalled();
    });

    it('should call onCollapseButtonClick', () => {
        const onCollapseButtonClickMock = jest.fn();
        const { getByTestId } = render(
            <Notification
                dataAttrs={dataAttrs}
                shortText="Short text"
                onCollapseButtonClick={onCollapseButtonClickMock}
            >
                Some text
            </Notification>,
        );

        fireEvent.click(getByTestId('collapseButton'));

        expect(onCollapseButtonClickMock).toBeCalled();
    });
});
