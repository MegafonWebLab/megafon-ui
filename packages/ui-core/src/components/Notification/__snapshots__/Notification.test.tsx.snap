// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> it renders Notification with close collapse 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text"
          >
            Короткий текст
          </div>
          <div
            className="mfui-notification__full-text"
          >
            Длинный текст
          </div>
        </div>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Подробнее
          <div
            className="mfui-notification__collapse-arrow"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification with close collpse after change isCollapseOpened 1`] = `
<Notification
  buttonText="Кнопка"
  className="notification"
  classes={
    Object {
      "container": "containerClass",
      "content": "contentClass",
      "root": "rootClass",
    }
  }
  hasCloseButton={false}
  href="href"
  icon={<test-file-stub />}
  isCollapseOpened={false}
  isColored={true}
  link="Ссылка"
  rel="noopener"
  shadowLevel="hover"
  shortText="Короткий текст"
  target="_blank"
  title="title"
  type="error"
>
  <Tile
    className="mfui-notification mfui-notification_type_error mfui-notification_colored notification rootClass"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    radius="rounded"
    shadowLevel="hover"
  >
    <div
      className="mfui-tile mfui-tile_shadow-color_light mfui-tile_radius_rounded mfui-tile_shadow_hover mfui-notification mfui-notification_type_error mfui-notification_colored notification rootClass"
      onClick={[Function]}
    >
      <div
        className="mfui-notification__container containerClass"
      >
        <div
          className="mfui-notification__icon-container"
        >
          <test-file-stub />
        </div>
        <div
          className="mfui-notification__content contentClass"
        >
          <div
            className="mfui-notification__text-container"
          >
            <span
              className="mfui-notification__title"
            >
              title
            </span>
            <div
              className="mfui-notification__text"
            >
              <div
                className="mfui-notification__short-text"
              >
                Короткий текст
              </div>
              <div
                className="mfui-notification__full-text"
              >
                Длинный текст
              </div>
            </div>
          </div>
          <div
            className="mfui-notification__bottom mfui-notification__bottom_has-button"
          >
            <div
              className="mfui-notification__bottom-block"
            >
              <Button
                className="mfui-notification__button"
                dataAttrs={
                  Object {
                    "root": undefined,
                  }
                }
                disabled={false}
                ellipsis={true}
                showLoader={false}
                sizeAll="small"
                sizeMobile="extra-small"
                theme="white"
              >
                <button
                  className="mfui-button mfui-button_type_primary mfui-button_theme_white mfui-button_size-all_small mfui-button_size-mobile_extra-small mfui-button_no-touch mfui-button_content-type_text mfui-notification__button"
                  disabled={false}
                  download={false}
                  onClick={[Function]}
                  type="button"
                >
                  <div
                    className="mfui-button__inner"
                  >
                    <div
                      className="mfui-button__content mfui-button__content_ellipsis"
                    >
                      <span
                        className="mfui-button__text mfui-button__text_ellipsis"
                      >
                        Кнопка
                      </span>
                    </div>
                  </div>
                </button>
              </Button>
            </div>
            <button
              className="mfui-notification__collapse-button"
              onClick={[Function]}
              type="button"
            >
              Подробнее
              <div
                className="mfui-notification__collapse-arrow"
              >
                <test-file-stub />
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </Tile>
</Notification>
`;

exports[`<Notification /> it renders Notification with custom close collapse title 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text"
          >
            Короткий текст
          </div>
          <div
            className="mfui-notification__full-text"
          >
            Длинный текст
          </div>
        </div>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Показать весь текст
          <div
            className="mfui-notification__collapse-arrow"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification with custom open collapse title 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text mfui-notification__short-text_hidden"
          >
            Короткий текст
          </div>
          <div
            className="mfui-notification__full-text mfui-notification__full-text_visible"
          >
            Длинный текст
          </div>
        </div>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Скрыть часть текста
          <div
            className="mfui-notification__collapse-arrow mfui-notification__collapse-arrow_close"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification with open collapse 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text mfui-notification__short-text_hidden"
          >
            Короткий текст
          </div>
          <div
            className="mfui-notification__full-text mfui-notification__full-text_visible"
          >
            Длинный текст
          </div>
        </div>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Свернуть
          <div
            className="mfui-notification__collapse-arrow mfui-notification__collapse-arrow_close"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification without link, when component has collapse 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_error mfui-notification_colored notification rootClass"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="hover"
>
  <div
    className="mfui-notification__container containerClass"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content contentClass"
    >
      <div
        className="mfui-notification__text-container"
      >
        <span
          className="mfui-notification__title"
        >
          title
        </span>
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text"
          >
            Короткий текст
          </div>
          <div
            className="mfui-notification__full-text"
          >
            Длинный текст
          </div>
        </div>
      </div>
      <div
        className="mfui-notification__bottom mfui-notification__bottom_has-button"
      >
        <div
          className="mfui-notification__bottom-block"
        >
          <Button
            className="mfui-notification__button"
            dataAttrs={
              Object {
                "root": undefined,
              }
            }
            disabled={false}
            ellipsis={true}
            showLoader={false}
            sizeAll="small"
            sizeMobile="extra-small"
            theme="white"
          >
            Кнопка
          </Button>
        </div>
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Подробнее
          <div
            className="mfui-notification__collapse-arrow"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders colored Notification with Props 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_error mfui-notification_colored notification rootClass"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="hover"
>
  <div
    className="mfui-notification__container containerClass"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content contentClass"
    >
      <div
        className="mfui-notification__text-container"
      >
        <span
          className="mfui-notification__title"
        >
          title
        </span>
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text"
          >
            Some test text
          </div>
        </div>
      </div>
      <div
        className="mfui-notification__bottom mfui-notification__bottom_has-button"
      >
        <div
          className="mfui-notification__bottom-block"
        >
          <Button
            className="mfui-notification__button"
            dataAttrs={
              Object {
                "root": undefined,
              }
            }
            disabled={false}
            ellipsis={true}
            showLoader={false}
            sizeAll="small"
            sizeMobile="extra-small"
            theme="white"
          >
            Кнопка
          </Button>
          <TextLink
            className="mfui-notification__link"
            dataAttrs={
              Object {
                "root": undefined,
              }
            }
            href="href"
            rel="noopener"
            target="_blank"
          >
            Ссылка
            <test-file-stub
              className="mfui-notification__link-arrow"
            />
          </TextLink>
        </div>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> renders component 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <div
          className="mfui-notification__text"
        >
          <div
            className="mfui-notification__short-text"
          >
            Some test text
          </div>
        </div>
      </div>
    </div>
  </div>
</Tile>
`;
